### DISCUSSION TITLE
How can I prevent the Traefik container publishing a port?

### DISCUSSION THREAD

**Discussion by trevorfoxsoft** [basecamp/kamal#343] [Q&A]
I'm running Traefik behind HAProxy and have got everything working well. HAProxy is talking to Traefik through it's container IP however, so there is no need for Traefik to bind to a port on the host. How can I tell it not to?

**Comment by scart88** [comment]
Did you tried ?

```yaml
  traefik:
    host_port: 8080
```

**Reply by trevorfoxsoft** [reply]
That's just telling it to run `docker --publish 8080:80` though, no? I want to not pass `--publish` at all.

**Comment by thr3a** [comment]
I succeeded in making Traefik not expose the port by setting this.

```yml
traefik:
  publish: false
```
